packages/core/src/app/ovm/deciders/and-decider.ts

```diff
@@ -0,0 +1,52 @@
+import {
+  Decider,
+  Decision,
+  ImplicationProofElement,
+  Property,
+} from '../../../types/ovm/decider.interface'
+
+export interface AndDeciderInput {
+  left: Property
+  leftWitness: any
+  right: Property
+  rightWitness: any
+}
+
+/**
+ * Decider that decides true iff both of the provided properties evaluate to true.
+ */
+export class AndDecider implements Decider {
+  public async decide(
+    input: AndDeciderInput,
+    witness: undefined
+  ): Promise<Decision> {
+    const [leftDecision, rightDecision] = await Promise.all([
+      input.left.decider.decide(input.left.input, input.leftWitness),
+      input.right.decider.decide(input.right.input, input.rightWitness),
+    ])
+
+    if (!leftDecision.outcome) {
+      return leftDecision
+    }
+    if (!rightDecision.outcome) {
+      return rightDecision
+    }
```

Comment:
Here I believe instead of only returning the decision which invalidated it, we need to return a new decision with this AND included as an implication proof element. This is so that when we guard the justification for our local decision includes the AND decision